sklearn/metrics/pairwise.py

```diff
@@ -53,14 +53,12 @@ def _return_float_dtype(X, Y):
     1. If dtype of X and Y is float32, then dtype float32 is returned.
     2. Else dtype float is returned.
     """
+    xp, _ = get_namespace(X, Y)
     if not issparse(X) and not isinstance(X, np.ndarray):
-        X = np.asarray(X)
+        X = xp.asarray(X)
 
     if Y is None:
         Y_dtype = X.dtype
-    elif not issparse(Y) and not isinstance(Y, np.ndarray):
-        Y = np.asarray(Y)
-        Y_dtype = Y.dtype
```

Comment:
Question, why delete this?

This is the cause of the `AttributeError: 'list' object has no attribute 'dtype'` and `AttributeError: 'tuple' object has no attribute 'dtype'` errors, for when `Y` is not an array.